gateway/handlers/forwarding_proxy.go

```diff
@@ -66,7 +87,7 @@ func buildUpstreamRequest(r *http.Request, url string) *http.Request {
 
 func forwardRequest(w http.ResponseWriter, r *http.Request, proxyClient *http.Client, baseURL string, requestURL string, timeout time.Duration) (int, error) {
 
-	upstreamReq := buildUpstreamRequest(r, baseURL+requestURL)
+	upstreamReq := buildUpstreamRequest(r, baseURL, requestURL)
```

Comment:
I had a previous question about whether this belongs in the generic code which forwards a request upstream. I feel like it belongs in a middleware handler which wraps this one, in that way we retain single-responsibility and the transformation is limited to the routes we add the middleware to. This would also allow one piece of code to cover the `/function/` and `/async-function/` routes.

Reply:
A good example might be the CallID middleware?

Reply:
Or the basic auth middleware